x-pack/plugin/sql/qa/src/main/resources/geo/geosql.csv-spec

```diff
@@ -214,3 +214,55 @@ SELECT ST_Distance(ST_WktToSql(NULL), ST_WktToSQL('POINT (-71 42)')) shape;
   shape:d
 null
 ;
+
+groupByGeometryType
+SELECT COUNT(*) cnt, ST_GeometryType(location) gt FROM geo GROUP BY ST_GeometryType(location);
+
+     cnt:l     |      gt:s
+15             |POINT
+;
+
+groupByEastWest
+SELECT COUNT(*) cnt, FLOOR(ST_X(location)/90) east FROM geo GROUP BY east ORDER BY east;
+
+     cnt:l     |     east:l
+3              |-2
+3              |-1
+4              |0
+5              |1
+;
+
+groupByNorthSouth
+SELECT COUNT(*) cnt, FLOOR(ST_Y(location)/45) north FROM geo GROUP BY north ORDER BY north;
+
+     cnt:l     |      north:l
+1              |-1
+9              |0
+5              |1
+;
+
+selectFilterByXOfLocation
+SELECT city, ST_X(shape) x, ST_Y(shape) y FROM geo WHERE ST_X(location) > 0 ORDER BY ST_Y(location);
+
+   city:s      |      x:d      |     y:d
```

Comment:
In this query I would also add as returned columns `ST_X(location)` and `ST_Y(location)` just to be easier to double check the results' correctness.